dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +326,293 @@ private FileFilter getNoBundleXMLFileFilter() {
+
+        final String bundleName = "testCreatePublisherFilterAllFilters";
+        Bundle bundle = new Bundle(bundleName, new Date(), null, adminUser.getUserId(),
+                (Boolean) filtersMap.get("forcePush"),filterKey);
+        APILocator.getBundleAPI().saveBundle(bundle);
+        bundle = APILocator.getBundleAPI().getBundleByName(bundleName);
+        Assert.assertNotNull(bundle);
+        Assert.assertEquals(filterKey,bundle.getFilterKey());
+
+        final PublisherFilter publisherFilter = publisherAPI.createPublisherFilter(bundle.getId());
+        Assert.assertNotNull(publisherFilter);
+        Assert.assertEquals(filtersMap.get("dependencies"),publisherFilter.isDependencies());
+        Assert.assertEquals(filtersMap.get("relationships"),publisherFilter.isRelationships());
+        Assert.assertEquals(filtersMap.get("forcePush"),bundle.isForcePush());
+        Assert.assertFalse(publisherFilter.acceptExcludeClasses("User"));//Will not be accepted since it's in the list
+        Assert.assertTrue(publisherFilter.acceptExcludeClasses("Template"));
+        Assert.assertFalse(publisherFilter.acceptExcludeDependencyClasses("Host"));//Will not be accepted since it's in the list
+        Assert.assertFalse(publisherFilter.acceptExcludeDependencyClasses("ContentType"));//Will not be accepted since it's in the list
+        Assert.assertTrue(publisherFilter.acceptExcludeDependencyClasses("Structure"));
+
+    }
+
+    /**
+     * This test creates a FilterDescriptor and a Bundle (with the created FilterDescriptor) and
+     * returns a PublisherFilter with the filters of the FilterDescriptor.
+     *
+     * In this case the FilterDescriptor has some of the possible filters.
+     */
+    @Test
+    public void test_createPublisherFilter_withSomeFilters()
+            throws DotDataException, DotSecurityException {
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final String filterKey = "TestCreatePublisherFilter.yml";
+        final List<Object> listExcludeClasses = new ArrayList();
+        listExcludeClasses.add("User");
+        listExcludeClasses.add("OSGI");
+        final Map<String, Object> filtersMap =
+                ImmutableMap
+                        .of("excludeClasses", listExcludeClasses, "dependencies", true, "forcePush", false,
+                                "relationships", true);
+        createFilterDescriptor(filterKey, "TestCreatePublisherFilterSomeFilters", true,
+                filtersMap);
+
+        final String bundleName = "testCreatePublisherFilterSomeFilters";
+        Bundle bundle = new Bundle(bundleName, new Date(), null, adminUser.getUserId(),
+                (Boolean) filtersMap.get("forcePush"), filterKey);
+        APILocator.getBundleAPI().saveBundle(bundle);
+        bundle = APILocator.getBundleAPI().getBundleByName(bundleName);
+        Assert.assertNotNull(bundle);
+        Assert.assertEquals(filterKey, bundle.getFilterKey());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)